modules/php/Cards/Rules/RuleGoalMill.php

```diff
@@ -2,29 +2,84 @@
 namespace Fluxx\Cards\Rules;
 
 use Fluxx\Game\Utils;
+use fluxx;
 
 class RuleGoalMill extends RuleCard
 {
   public function __construct($cardId, $uniqueId)
   {
     parent::__construct($cardId, $uniqueId);
 
-    $canBeUsedByPlayer = true;
-
     $this->name = clienttranslate("Goal Mill");
     $this->subtitle = clienttranslate("Free Action");
     $this->description = clienttranslate(
       "Once during your turn, discard as many of your Goal cards as you choose, then draw that many cards."
     );
   }
 
+  public $interactionNeeded = "handCardsSelection";
+
+  public function canBeUsedInPlayerTurn($player_id)
+  {
+    $alreadyUsed = !Utils::playerHasNotYetUsedGoalMill();
```

Comment:
This method is only used here, so it's weird to define it as "has_not_yet" and then do a `!`.

Reply:
I see that's because in the other cases, "not yet" makes more sense. Ignore me then :-)

Reply:
Yes, exactly, just  to be consistent in all the "playerTurn' game state value checks.